source/topic_tree.c

```diff
@@ -368,8 +368,13 @@ static void s_topic_tree_action_commit(struct topic_tree_action *action, struct
                 action->node_to_update->topic = action->topic;
             }
             if (action->topic_filter) {
-                action->node_to_update->topic_filter = action->topic_filter;
-                action->node_to_update->owns_topic_filter = true;
+                if (action->node_to_update->owns_topic_filter && action->node_to_update->topic_filter) {
```

Comment:
I don't see how owns_topic_filter is safe to read then write here. Maybe I haven't fully understood the context, but it appears this is the point in code to authoritatively set the flag and not read its current value to make a decision.

Reply:
yeah, this is a weird situation. I did this since when there are two pending actions with the same topic filter in one commit. The idea was if the `node_to_update` already owned the `topic_filter`, the other insertion of the same `topic_filter` will be cleaned up in [aws_mqtt_topic_tree_transaction_insert](https://github.com/awslabs/aws-c-mqtt/pull/137/files#diff-2e434658c1666944abda6141f0b9400dL673). 
However, if the old action has not been committed yet. The old node will not own `topic_filter` until the commit called, the new string will not be cleaned up in `aws_mqtt_topic_tree_transaction_insert`. Possible memory leak here in that case.
It should be safe to read the `node_to_update` here, since the node will be zero out when it's created in ADD case, and the old node is always valid in UPDATE case.